Update EIP-5450: Add JUMPF simplification (#6101)
ethereum/EIPs: EIPS/eip-5450.md

Stack overflow check, on the other hand, is still required at run-time, because function execution can start at arbitrary (i.e. known only at run-time) stack height at `CALLF` instruction of a caller (i.e. each execution can be in arbitrary inner call frame). Verification algorithm examines only stack height changes relative to starting stack height of the function.

> #### JUMPF changes

Reviewer: Minor nit: much of these section could live in rationale under something like "Reducing gas cost of JUMPF".

> In case function pushed on the stack more items than inputs required by the jumped-to function, previously defined `JUMPF` instuction behaviour included removing extra items:

Reviewer: ```suggestion
In case a function pushed more items to the stack than is required as inputs by the jumped-to function, the previously defined `JUMPF` instruction behaviour was to remove the extra items:
```

Reviewer: might be slightly clearer - also typo on "instruction"

> 3. If data stack has more than `caller_stack_height + type[code_section_index].inputs` items, discards the items between `caller_stack_height` and top `type[code_section_index].inputs`, so that there are exactly `caller_stack_height +  type[code_section_index].items` items left.

> Given new deploy-time guarantees of function execution never underflowing its stack frame, `JUMPF` instruction can check only that there is enough items for input arguments of the callee on the stack, without making sure there is exactly `inputs` items and not more. 

Reviewer: ```suggestion
Given the new deploy-time guarantee of no function underflowing its stack frame, `JUMPF` instruction can check only that there is enough items for input arguments of the callee on the stack, without making sure there is exactly `inputs` items and not more. 
```

> Therefore previously defined behavior of discarding extra items is removed, and only the check for enough inputs is done:

Reviewer: ```suggestion
Therefore, the previously defined behavior of discarding extra items is removed, and only the check for enough inputs is done:
```
